source/common/upstream/thread_aware_lb_impl.h

```diff
@@ -28,6 +29,43 @@ class ThreadAwareLoadBalancerBase : public LoadBalancerBase, public ThreadAwareL
   };
   using HashingLoadBalancerSharedPtr = std::shared_ptr<HashingLoadBalancer>;
 
+  /**
+   * Class for consistent hashing load balancer (CH-LB) with bounded loads.
+   * It is common to both RingHash and Maglev load balancers, because the logic of selecting the
+   * next host when one is overloaded is independent of the CH-LB type.
+   */
+  class BoundedLoadHashingLoadBalancer : public HashingLoadBalancer {
+  public:
+    BoundedLoadHashingLoadBalancer(HashingLoadBalancerSharedPtr hashing_lb_ptr,
+                                   NormalizedHostWeightVector normalized_host_weights,
```

Comment:
Yep this looks right. One way to force the caller to move the parameter to specify a rvalue ref (`NormalizedHostWeightVector&&`), but I don't think that's necessary here since this is an implementation specific class and not a public interface that others might be using.

Reply:
Got it. Thanks.